### PR TITLE
DOC: interpolate: detail `interp1d` recommended replacements

### LINKED ISSUE #21571: DOC: interpolate.interp1d: clarify status and alternatives
### Is your feature request related to a problem? Please describe.

Documentation now states that `scipy`'s `interp1d` is deprecated https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html . However, the main solution pointed to (`numpy`s `interp`) is not providing as much functionality. For example:

- in `scipy.interp1d`, I can build the interpolator class, and then call it several times on different arrays of interpolation points. I guess if the interpolator class is well implemented that it means that I may factor all "pre-processing and preparation" if there is any needed in `scipy.interp1d`, which may be quite a bit more efficient than with `numpy.interp` API which will need to do all of this again and again depending on use cases (?)

- `scipy.interp1d` supports a lot more functionalities, because of the flexible `kind` argument. For example there are cases when one needs nearest interpolation. This is not easily doable as far as I know with the `numpy.interp` function.

It is perfectly fine to me if 1d interpolation is deprecated from `scipy`and moved to `numpy`, but in this case it would be nice to have "at least as much and as good" functionality :) .

I can open a twin issue on `numpy`'s repo :) .

### Describe the solution you'd like.

Keep supporting `scipy`'s 1d interpolation until `numpy`'s one is offering at least as much / good functionality.

### Describe alternatives you've considered.

`numpy`'s interpolation, as recommended by the documentation.

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by ev-br**
#### Reference issue

closes https://github.com/scipy/scipy/issues/21571

#### What does this implement/fix?

Add a doc page to detail recommended replacements for `interp1d`. Also try stressing that interp1d is not deprecated, as users seem to read "legacy" as "deprecated", gh-21571 is a typical but definitely not the only example.

While at it, also address a concern about `interp1d` vs `np.interp` which shows once in a while (basically, `np.interp` is a bit more limited). To this end, mention more prominently the equivalent `make_interp_spline` incantation.

### CODE REVIEW

File: doc/source/tutorial/interpolate/1D.rst [hunk 1]
```diff
@@ -339,6 +341,55 @@ same data as in the previous example:
...
+   >>> x_new = np.linspace(0, 7, 101)    # input points
+   >>> x_bds = x[:-1] / 2.0 + x[1:] / 2.0   # halfway points
+   >>> idx = np.searchsorted(x_bds, x_new, side='left')
+   >>> idx = np.clip(idx, 0, len(x) - 1)  # clip the indices so that they are within the range of x indices.
+   >>> import matplotlib.pyplot as plt
+   >>> plt.plot(x, y, 'o')
+   >>> plt.plot(x_new, y[idx], '--')
+   >>> plt.show()
+
+Other variants are similar, see the :ref:```interp1d`` source code <https://github.com/scipy/scipy/blob/v1.14.1/scipy/interpolate/_interpolate.py#L486>`
```
Comment (j-bowhay) [inline doc/source/tutorial/interpolate/1D.rst:390]: This doesn't see to be rendering correctly

File: doc/source/tutorial/interpolate/1D.rst [hunk 2]
```diff
@@ -339,6 +341,55 @@ same data as in the previous example:
...
+
+The default recommendation is to use `numpy.interp` function. Alternatively,
+you can use linear splines, ``make_interp_spline(x, y, k=1)``, see
+:ref:`this section for a discussion<tutorial-interpolate_linear_spline>`.
+
+
+Spline interpolators, ``kind="quadatic"`` or ``"cubic"``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Under the hood, ``interp1d`` delegates to ``make_interp_spline``, so we recommend
```
Comment (j-bowhay) [inline doc/source/tutorial/interpolate/1D.rst:366]:
```suggestion
Under the hood, `interp1d` delegates to `make_interp_spline`, so we recommend
```

**Review by j-bowhay — COMMENTED** [review]
Thanks! One minor rendering issue and a optional change

File: doc/source/tutorial/interpolate/1D.rst (hunk 2 above)
Comment (ev-br) [reply doc/source/tutorial/interpolate/1D.rst]: Thanks, indeed. Fixed and force-pushed, both this and the other suggestion.

File: doc/source/tutorial/interpolate/1D.rst [hunk 3]
```diff
@@ -339,6 +341,52 @@ same data as in the previous example:
...
+Recommended replacements for `interp1d` modes
+---------------------------------------------
+
+As mentioned, `interp1d` class is *legacy*: we have no plans to remove it; we
+are going to keep supporting its existing usages; however
+we believe there are better alternatives which we recommend using in new code.
+
+Here we list specific recommendations, depending on the interpolation ``kind``.
+
+**Linear interpolation, ``kind="linear"``**
```
Comment (j-bowhay) [inline doc/source/tutorial/interpolate/1D.rst:355]: Hmm it also looks like monospace doesn't render correctly within bold

Comment (ev-br) [reply doc/source/tutorial/interpolate/1D.rst]: Grrr, third time the charm, hopefully! The current version (again, fixed and force-pushed) renders OK locally, let's see if CircleCI concurs. Sorry for the back-and-forth.

Comment (ev-br) [reply doc/source/tutorial/interpolate/1D.rst]: > let's see if CircleCI concurs

... and it does.

**Review by j-bowhay — APPROVED** [review]
Thanks!

**Pull request merged**